### PR TITLE
python-toml: add python@3.12

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

### CODE REVIEW

File: Formula/p/python-toml.rb [hunk 1]
```diff
@@ -18,23 +18,31 @@ class PythonToml < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "a4af3afd961e422d62ca04300144dd06e722e617b8ca90d0e66177980d306c66"
   end
 
-  depends_on "python@3.11"
+  depends_on "python-setuptools" => :build
+  depends_on "python@3.11" => [:build, :test]
+  depends_on "python@3.12" => [:build, :test]
```
Comment (bayandin) [inline Formula/p/python-toml.rb:23]: Shouldn't we install it for all pythons that we have? Well, maybe from 3.9

Comment (branchvincent) [reply Formula/p/python-toml.rb:23]: I think this formula is mainly intended as a library consumed by other formulae, and so my thinking is to only install for the pythons used by those dependents. That also makes it easier when we deprecate/disable `python@3.9` etc

But open to other opinions

Comment (branchvincent) [reply Formula/p/python-toml.rb:23]: Gotcha, btw #149418 swaps six's 3.9 for 3.12. I can add 3.10 here too to be consistent

Comment (bayandin) [reply Formula/p/python-toml.rb:23]: Ah, nice! Let's then just add `python@3.10` in this PR I think

**Review by bayandin — APPROVED** [review]

**Pull request merged**